### PR TITLE
basic graph functionality added, some unit tests, maximal non-branchi…

### PR DESCRIPTION
**Pull request by gwarmstrong**
…ng-paths still needed

### CODE REVIEW

**Comment by gwarmstrong** [comment]
maximal non branching paths does not yet capture cycles well

**Comment by gwarmstrong** [comment]
> maximal non branching paths does not yet capture cycles well

Fixed by 
a6a5085 and ee938c7

File: TesasSemble/graph.py [hunk 1]
```diff
@@ -1,44 +1,135 @@
...
+			start_node = edge.node_a
+			to_node = edge.node_b
+			while to_node != start_node:
+				out_edge = self.nodes[to_node]['out_edges'][0]
+				non_branching_path.append(out_edge)
+				cycle_edges.remove(out_edge)
+				visited_edges.add(out_edge)
+				to_node = out_edge.node_b
+				if len(cycle_edges) == 0:
+					break
```
Comment (gwarmstrong) [inline TesasSemble/graph.py]: we don't need lines 52 and 53

**Review by gwarmstrong — COMMENTED** [review]
These lines are not neccesary for the algorithm

File: TesasSemble/graph.py (hunk 1 above)
Comment (gwarmstrong) [reply TesasSemble/graph.py]: Resolved by b1795ad94ae637bf794ea9032c5b4c8457767acc

**Review by earmingol — APPROVED** [review]
Tests passed Travis CI

**Pull request merged**